test/Microsoft.AspNetCore.Routing.Tests/ConstraintMatcherTest.cs

```diff
@@ -60,8 +60,8 @@ public void MatchFail_LogsCorrectData()
 
             // Assert
             Assert.Empty(sink.Scopes);
-            Assert.Single(sink.Writes);
-            Assert.Equal(expectedMessage, sink.Writes[0].State?.ToString());
+            var write = Assert.Single(sink.Writes);
+            Assert.Equal(expectedMessage, write.State?.ToString());
```

Comment:
User your new formatted message property.

Reply:
ugh... I'll have to use the feature branch.

Reply:
which involves building all the transitive repos that use logging...

Reply:
It's on you conscious then.

Reply:
I have no conscience.